demos/README.md

```diff
@@ -0,0 +1,103 @@
+
+## MQTT examples
+
+Example directory: `mqtt`
+
+### **mqtt_demo_basic_tls**
+
+The example is single threaded and uses statically allocated memory;
+it uses QOS2 and therefore implements a retransmission mechanism
+for Publish messages. Retransmission of publish messages are attempted
+when a MQTT connection is established with a session that was already
+present. All the outgoing publish messages waiting to receive PUBREC
+are resent in this demo. In order to support retransmission all the outgoing
+publishes are stored until a PUBREC is received.
+
+### **mqtt_demo_mutual_auth**
+
+The example is single threaded and uses statically allocated memory;
+it uses QOS1 and therefore implements a retransmission mechanism
+for Publish messages. Retransmission of publish messages are attempted
+when a MQTT connection is established with a session that was already
+present. All the outgoing publish messages waiting to receive PUBACK
+are resent in this demo. In order to support retransmission all the outgoing
+publishes are stored until a PUBACK is received.
+
+### **mqtt_demo_plaintext**
+
+The example shown below uses MQTT APIs to send and receive MQTT packets
+over the TCP connection established using POSIX sockets.
+The example is single threaded and uses statically allocated memory;
+it uses QOS0 and therefore does not implement any retransmission
+mechanism for Publish messages. This example runs forever, if connection to
+the broker goes down, the code tries to reconnect to the broker with exponential
+backoff mechanism.
+
+### **mqtt_demo_serializer**
+
+Demo that shows use of the MQTT serializer / deserializer API
+to build an ultra-light weight MQTT client. This shows that a lighter weight
+MQTT client can be developed without using the higher-level
+MQTT API of core_mqtt.h, but just the serializer / deserializer API.
+The core_mqtt_serializer.h API lets user to serialize and
+deserialize MQTT messages into a user provided buffer.
+This API allows use of a statically allocated buffer.
+
+### **mqtt_demo_subscription_manager**
+
+Demo of an MQTT application that subscribes to multiple topic filters using a
+subscription manager to manage multiple subscriptions, register different
+callbacks for each subscription, and handle wildcard topics. It establishes
+a TLS connection with server-only authentication, and communicates at
+QoS 1 level with the broker.
+
+## Device shadow examples
+
+Example directory: `shadow/shadow_demo_main`
+
+### shadow/shadow_demo_main
+
+Demo for showing how to use the Device Shadow library's API. This version
+of Device Shadow API provide macros and helper functions for assembling MQTT topics
+strings, and for determining whether an incoming MQTT message is related to the
+device shadow. The Device Shadow library does not depend on a MQTT library,
+therefore the code for MQTT connections are placed in another file (shadow_demo_helpers.c)
+to make it easy to read the code using Device Shadow library.
```

Comment:
This paragraph seems to have more information about the Device Shadow library instead of the demo application. 
More information about the Shadow demo can be obtained from here: https://docs.aws.amazon.com/freertos/latest/lib-ref/embedded-csdk/202009.00/lib-ref/docs/doxygen/output/html/shadow_demo_main.html